### PR TITLE
LectureClassification Enum 활용 및 GET /v1/tags/main/{id}/evaluations 에 '교양' 조건 적용

### PR DESCRIPTION
**Pull request by davin111**
related issues: https://draft-waffle.atlassian.net/browse/SNUTT-246

# Major Changes
## 1. Lecture 및 SemesterLecture 에서 classification 을 위해 String 대신 enum LectureClassification 활용
- 관련 코드 수정
- GET /v1/lectures 에서 QueryDSL 사용된 부분에서 EnumPath 관련 함수 추가 cc. @Hank-Choi 

## 2. GET /v1/tags/main/{id}/evaluations
- 기획 스펙과 달리 '교양'에 한정하지 않던 API 수정

### CODE REVIEW

File: src/test/kotlin/com/wafflestudio/snuttev/scheduler/SnuttLectureSyncJobSliceTest.kt [hunk 1]
```diff
@@ -40,7 +40,7 @@ class SnuttLectureSyncJobSliceTest(
     ): SnuttSemesterLecture {
         return SnuttSemesterLecture(
             id = "",
-            classification = "",
+            classification = "교양",
```
Comment (YeonghyeonKO) [inline src/test/kotlin/com/wafflestudio/snuttev/scheduler/SnuttLectureSyncJobSliceTest.kt:43]: 꼼꼼한 테스트 부탁드립니다

File: src/main/kotlin/com/wafflestudio/snuttev/DataLoader.kt [hunk 2]
```diff
@@ -62,7 +63,7 @@ class DataLoader(
                 credit = 4,
                 academicYear = "3학년",
```
Comment (YeonghyeonKO) [inline src/main/kotlin/com/wafflestudio/snuttev/DataLoader.kt:64]: academicYear는 하드코딩으로 관리하는 이유가 있나요?

**Review by YeonghyeonKO — COMMENTED** [review]
LGTM

File: src/test/kotlin/com/wafflestudio/snuttev/scheduler/SnuttLectureSyncJobSliceTest.kt (hunk 1 above)
Comment (davin111) [reply src/test/kotlin/com/wafflestudio/snuttev/scheduler/SnuttLectureSyncJobSliceTest.kt:43]: ㅋㅋㅋㅋㅋ GitHub Actions 가 이미 해줬어요

File: src/main/kotlin/com/wafflestudio/snuttev/DataLoader.kt (hunk 2 above)
Comment (davin111) [reply src/main/kotlin/com/wafflestudio/snuttev/DataLoader.kt:64]: 아 이것도 물론 enum 으로 써도 좋을 거 같긴 한데, 추후 적용하겠습니다.

File: src/main/kotlin/com/wafflestudio/snuttev/domain/lecture/model/SemesterLecture.kt [hunk 3]
```diff
@@ -47,7 +48,7 @@ data class SemesterLectureWithLecture(
 
     var category: String,
 
-    var classification: String,
+    var classification: LectureClassification,
```
Comment (Hank-Choi) [inline src/main/kotlin/com/wafflestudio/snuttev/domain/lecture/model/SemesterLecture.kt:51]: 여기는 @Convert 필요 없는거야?

File: src/test/kotlin/com/wafflestudio/snuttev/scheduler/SnuttLectureSyncJobSliceTest.kt (hunk 1 above)
Comment (Hank-Choi) [reply src/test/kotlin/com/wafflestudio/snuttev/scheduler/SnuttLectureSyncJobSliceTest.kt:43]: 이건 왜 이렇게 바꾼거지

Comment (Hank-Choi) [reply src/test/kotlin/com/wafflestudio/snuttev/scheduler/SnuttLectureSyncJobSliceTest.kt:43]: 빈칸일 때 에러나서?

File: src/main/kotlin/com/wafflestudio/snuttev/domain/lecture/model/Lecture.kt [hunk 4]
```diff
@@ -41,9 +42,21 @@ enum class LectureClassification(@get:JsonValue val value: String) {
     READING_AND_RESEARCH("논문"),
     TEACHING_CERTIFICATION("교직"),
     GRADUATE("대학원"),
-    CORE_SUBJECT("공통"),
+    CORE_SUBJECT("공통");
+
+    companion object {
+        private val mapping = values().associateBy { e -> e.value }
+
+        fun customValueOf(classification: String): LectureClassification? = mapping.getOrDefault(classification, null)
```
Comment (Hank-Choi) [inline src/main/kotlin/com/wafflestudio/snuttev/domain/lecture/model/Lecture.kt:50]: classification이 새로 추가될 수도 있는거 아닌가?
뭔가 여기 없는 값일 때 null 튀어나가면 이상할 듯 한데

**Review by Hank-Choi — COMMENTED** [review]
나중에 내가 말한 전공 필터 추가를 위한 준비작업도 같이 있는 것 같아 매우 좋습니다. 코멘트 몇가지 답변 부탁드려요~~

File: src/test/kotlin/com/wafflestudio/snuttev/scheduler/SnuttLectureSyncJobSliceTest.kt (hunk 1 above)
Comment (davin111) [reply src/test/kotlin/com/wafflestudio/snuttev/scheduler/SnuttLectureSyncJobSliceTest.kt:43]: 맞아, Enum 중에 매핑되는 게 없는데 SemesterLecture, Lecture 의 classification 은 null 이 되면 안 되어서

File: src/main/kotlin/com/wafflestudio/snuttev/domain/lecture/model/SemesterLecture.kt (hunk 3 above)
Comment (davin111) [reply src/main/kotlin/com/wafflestudio/snuttev/domain/lecture/model/SemesterLecture.kt:51]: @Hank-Choi 응 여기는 알아서 해주더라구

File: src/main/kotlin/com/wafflestudio/snuttev/domain/lecture/model/Lecture.kt (hunk 4 above)
Comment (davin111) [reply src/main/kotlin/com/wafflestudio/snuttev/domain/lecture/model/Lecture.kt:50]: 처음에는 매핑되는 거 없으면 여기서 터지고 null 반환 안 하게 나도 짜뒀는데, 검색 쪽의 EnumPath 의 isIn 구현하다보니까, tags 가 아무 String 이나 들어올 수 있어서 이렇게 바꿨어.

물론 학교에서 classification 갑자기 새로운 걸 추가할 수 있는데, 그 경우엔 SnuttLectureSyncJobService 보면 !! 처리 해놔서 거기서 터져서 알 수 있을 거야. @Hank-Choi
